packages/client/src/runtime/core/engines/library/LibraryEngine.ts

```diff
@@ -113,6 +124,41 @@ export class LibraryEngine implements Engine<undefined> {
     this.libraryInstantiationPromise = this.instantiateLibrary()
   }
 
+  private wrapEngine(engine: QueryEngineInstance) {
+    return {
+      applyPendingMigrations: engine.applyPendingMigrations?.bind(engine),
+      commitTransaction: this.withRequestId(engine.commitTransaction?.bind(engine)),
+      connect: this.withRequestId(engine.connect?.bind(engine)),
+      disconnect: this.withRequestId(engine.disconnect?.bind(engine)),
+      dmmf: engine.dmmf?.bind(engine),
+      metrics: engine.metrics?.bind(engine),
+      query: this.withRequestId(engine.query?.bind(engine)),
+      rollbackTransaction: this.withRequestId(engine.rollbackTransaction?.bind(engine)),
+      sdlSchema: engine.sdlSchema?.bind(engine),
+      startTransaction: this.withRequestId(engine.startTransaction?.bind(engine)),
+      trace: engine.trace?.bind(engine),
+    }
```

Comment:
Why does this use `?` operators (e.g., in `engine.connect?.bind(engine)`)?

`QueryEngineInstance` has type

```typescript
export type QueryEngineInstance = {
  connect(headers: string, requestId: string): Promise<void>
  disconnect(headers: string, requestId: string): Promise<void>
 // ...
}
```

Reply:
The type is not exactly correct, not all of them are actually present in all engines :(
For example `applyPendingMigrations` is only present in c-abi engine. `dmmf` and `sdlSchema` are not everywhere IIRC. I just put `?` everywhere instead of figuring it out properly and fixing the type, thanks for calling this out, it's just lazy coding. I'll fix it.

Reply:
9debd4a54dab59b97bb148e34d5ce74d49bb8e91